core/src/main/java/com/amplifyframework/logging/LogEntry.java

```diff
@@ -16,26 +16,24 @@
 package com.amplifyframework.logging;
 
 import android.util.Log;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
 
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
+import java.util.Objects;
 
 /**
  * A representation of a log.
  */
 public final class LogEntry {
```

Comment:
You could bundle together all of your code for the dev menu into `core/src/main/java/com/amplifyframework/devmenu`. This `LogEntry` is more a thing that's used by the dev menu, I'd say, than it is part of the core spec of the logger.

`core/src/main/java/com/amplifyframework/logging` is meant to be a core spec for the logger category.

It does also include the default Android implementations of the plugin and logger. Which isn't 100% architecturally pure, true. In other categories, those plugins would be separate Gradle modules entirely.

Reply:
Would PersistentLogStoragePlugin and PersistentLogger go in the devmenu package or logging package?

Reply:
/md One could argue either way. I'd say _devmenu_, since its the only consumer we currently expect for the plugin.

If we find another use for it, we can always move it back to the more general location, later.